File: mobile/lib/models/pyxis.dart
class Pyxis {
  String? createdAt;
  String? updatedAt;
  String? id;
  String? label;

  Pyxis({this.createdAt, this.id, this.label, this.updatedAt});

  Pyxis.fromJson(Map<String, dynamic> json) {
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
    id = json['id'];
    label = json['label'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['created_at'] = createdAt;
    data['updated_at'] = updatedAt;
    data['id'] = id;
    data['label'] = label;
    return data;
  }
}

File: backend/internal/domain/entity/pyxis.go
package entity

import (
	"time"
)

type PyxisRepository interface {
	CreatePyxis(pyxis *Pyxis) (*Pyxis, error)
	FindPyxisById(id string) (*Pyxis, error)
	FindAllPyxis() ([]*Pyxis, error)
	UpdatePyxis(pyxis *Pyxis) (*Pyxis, error)
	DeletePyxis(id string) error
}

type Pyxis struct {
	ID        string    `json:"id" db:"id"`
	Label     string    `json:"label" db:"label"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
	UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
}

func NewPyxis(label string) *Pyxis {
	return &Pyxis{
		Label:     label,
	}
}
